plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTable.java

```diff
@@ -300,7 +300,7 @@ public static Object convert(Object value, Type type)
             if (value instanceof Long) {
                 return value;
             }
-            throw new IllegalArgumentException("Unexpected type : " + type);
+            throw new IllegalArgumentException("Unexpected type of value : " + value);
```

Comment:
More accurate commit message would be `Fix statistics computation for nested types`